pkg/precheck/precheck.go

```diff
@@ -0,0 +1,123 @@
+package precheck
+
+import (
+	"sync"
+
+	"github.com/juju/errors"
+	"github.com/ngaut/log"
+)
+
+// PreChecker is interface about precheck.
+type PreChecker interface {
+	Name() string
+	PreCheck() *PreCheckResult
+}
+
+// PreCheckState is state of precheck
+type PreCheckState string
+
+const (
+	PreCheckState_Success PreCheckState = "success"
+	PreCheckState_Failure PreCheckState = "fail"
+	PreCheckState_Warning PreCheckState = "warn"
```

Comment:
how about
```
StateSuccess PreCheckState = "success"
StateFailure PreCheckState = "fail"
StateWarning PreCheckState = "warn"
```
i think we don't need the variable name starts with the package name

and below ditto, `PreCheckResults`, `PreCheckResultSummary`, `PreCheckResults`